### PR TITLE
[asio-grpc] Update to 1.7.0

### PR DESCRIPTION
**Pull request by Tradias**
Updates asio-grpc to v1.7.0.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes

### CODE REVIEW

File: ports/asio-grpc/portfile.cmake [hunk 1]
```diff
@@ -15,13 +15,12 @@ vcpkg_check_features(
 vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
     OPTIONS ${FEATURE_OPTIONS}
+        -DASIO_GRPC_CMAKE_CONFIG_INSTALL_DIR=share/asio-grpc
 )
 
 vcpkg_cmake_install()
 
-vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/asio-grpc)
```
Comment (FrankXie05) [inline ports/asio-grpc/portfile.cmake:22]: This method depends on `vcpkg-cmake-config`, please delete it together.
    ````
    {
      "name": "vcpkg-cmake-config",
      "host": true
    }
    ````

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Comment by Tradias** [comment]
@FrankXie05 yes thank you. I have updated several occurrences of asio-grpc with PROJECT_NAME. I won't create a patch release for it though, so we can go ahead with this pr.

**Review by FrankXie05 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
@FrankXie05 It is not generally an improvement to mass-replace all instances of `asio-grpc` with `${PROJECT_NAME}`. It's fine either way and we should avoid excessive changes one way or another.

File: ports/asio-grpc/portfile.cmake [hunk 2]
```diff
@@ -15,13 +15,12 @@ vcpkg_check_features(
...
     OPTIONS ${FEATURE_OPTIONS}
+        -DASIO_GRPC_CMAKE_CONFIG_INSTALL_DIR=share/asio-grpc
 )
 
 vcpkg_cmake_install()
 
-vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/asio-grpc)
-
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug" "${CURRENT_PACKAGES_DIR}/lib")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
```
Comment (ras0219-msft) [inline ports/asio-grpc/portfile.cmake:23]:
```suggestion
vcpkg_cmake_config_fixup()
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
```

Even for 'header only' libraries, `vcpkg_cmake_config_fixup()` is still important to patch out any potential absolute paths that might get added to the files.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Thanks for the update!

**Pull request merged**